server/src/com/cloud/network/element/VirtualRouterElement.java

```diff
@@ -988,6 +1016,21 @@ public boolean addDhcpEntry(final Network network, final NicProfile nic, final V
         return result;
     }
 
+    protected boolean deleteDhcpSupportForSubnet(Network network, Network.Service service) throws ResourceUnavailableException {
+        if (canHandle(network, service)) {
+            final List<DomainRouterVO> routers = _routerDao.listByNetworkAndRole(network.getId(), Role.VIRTUAL_ROUTER);
+            if (routers == null || routers.size() == 0) {
```

Comment:
Minor nit: Please consider using ``isEmpty`` rather than a size check as it is clearer intent and more idiomatic.